### DISCUSSION TITLE
BRouter (offline) routing in Villages/Cities

### DISCUSSION THREAD

**Discussion by micclass62** [devemux86/cruiser#590] [Q&A] [answered]
Hello,

recently came back from a 2000km+ Motorbike tour through Switzerland and the French Alps. Was using Cruiser (4.0.2) with BRouter (offline) for the routing. 

Coming from Kurviger2 (which was the optimal solutions for myself) Cruiser is working well too. Nothing to complain about navigation at all. Rerouting (due to blocked roads) worked even better now with offline BRouter (in the past this was often not working due to missing mobile connection in rural areas). 

The only issue I do have: For the routing I do use a "Curvy" profile with:

Autobahnen: 4/5, Fernstrassen: 3/5, Hauptstrassen 2/5, Sekundärstrassen: Aus, Tertiätstrassen: Aus, Nicht klassifizierte Strassen: 3/5, Wohnstrassen: 4/5

Outside of Cities and Villages this leads to routes that fit my expectations quite well. Only in very rare cases that are small "deviations" away from the road I am on coming back a few hundert meters later. Typically I can see such a situation and avoid it.

Unfortunately inside villages and cities I do get a lot of routing through very small roads (even sometimes through courtyards) and housing areas. This is not what I do want. 
For the - rare - cases that I do pass through larger cities (I try to avoid doing that) I can manually handle the situation by adding shapingpoints in front and after the city and change the profile to fastest in between (thanks for that feature!). But for the smaller villages this would be way to much work to be handled manually.

Question: Is there a way to avoid going through very small road inside villages (and cities) automatically? Outside I still want to use the curvy profile, but inside I would want to switch to fastest. Any chance to get such a behaviour?

Cheers,

Michael

**Comment by devemux86** [comment, marked as answer]
Hi @micclass62 thanks for the kind words!

It's a little complicated to do it automatically without compromises.
(especially for all kind of populated areas, like villages)

If want a specific route profile (e.g. fastest) in a specific area,
better to set 2 waypoints before and after the area and set the route profile for that route section.

To avoid specific areas we are working on a new feature to freely define **no-go** areas in route planning:
- https://github.com/devemux86/cruiser/discussions/538

The first implementation is already available for testing:
- https://github.com/devemux86/cruiser/discussions/544

**Reply by Leeb64** [reply]
You could reduce the avoidance factor for residential streets (avoid urban factor) from 1 (->for 4/5) to 0.2 or 0.1.

Turning on "consider Town" helps!

**Reply by devemux86** [reply]
@Leeb64 

4/5 is 5, 5/5 is 1, so can use 5/5 or extra curvy route.

**Reply by Leeb64** [reply]
That's right. Off is 0!
I thought 4/5 would be 1...and 5/5 would be 0.
Then 5/5 should be reduced to ~ 0.2.

**Reply by devemux86** [reply]
0 is absolute avoid and can lead to failed routing and users not understanding the reason, so it's better to avoid it.

**Reply by Leeb64** [reply]
Normally, when “Avoid Cities” is activated, things like driving on small streets like this shouldn’t happen.
But you could also reduce the "Avoid Urban/Residential Streets" factor to below "1" at 5/5.
I just ran a few quick tests and I think 0.5 is still OK.(Especially at “avoid urban factor”)
Increasing the speed also helps

**Reply by devemux86** [reply]
I think that BRouter's avoid cities currently work with minimum population 50 k people,
so we need much less population to be really useful.

Anyway, these tags sometimes work or not, so better to use the common avoid road types and [no-go areas](https://github.com/devemux86/cruiser/discussions/538).

Their documentation is [here](https://github.com/abrensch/brouter/blob/master/docs/developers/environmental_considerations_and_pseudo_tags.md).

**Comment by Leeb64** [comment]
![grafik](https://github.com/devemux86/cruiser/assets/120722053/630ac7b5-bdd6-4049-b4c9-70374376c946)
consider Town:OFF
![grafik](https://github.com/devemux86/cruiser/assets/120722053/da5fa982-4d45-4f92-8a5f-e4ee3c31f2f8)
consider Town:ON
No city with more than 50,000 inhabitants nearby

**Reply by WolfgangZ42** [reply]
@Leeb64 
Could you please help me? What URL and what vehicle did you use to get the above screens? 
I tried to get such selection of options and failed:
![2024-07-07 13_37_54-BRouter Web Client](https://github.com/devemux86/cruiser/assets/121806238/4249ae00-daa9-49a7-b788-6f9dd9e42983)
You see, I took "Treckingrad" - Auto and Moped didn't show a similar collection ... 
Thank you in advance!

**Reply by Leeb64** [reply]
Ausgangsprofil ist driving-cruiser:
[driving-cruser.brf.txt](https://github.com/user-attachments/files/16119827/driving-cruser.brf.txt)

Ich habe mir 2 Profile erstellt.
[Landstraße 2+3 .brf.txt](https://github.com/user-attachments/files/16119671/Landstrasse.2%2B3.brf.txt)
[Landstraße 3+4 .brf .txt](https://github.com/user-attachments/files/16119672/Landstrasse.3%2B4.brf.txt)
(Das "txt" löschen.Braucht man nur zum Hochladen hier.)
Das erste bevorzugt Strassen 2. + 3. "Kategorie",das zweite Strassen 3. + 4. "Kategorie"
Beide versuchen Ortschaften+Städte zu umfahren falls möglich-es werden zum Teil auch mautfreie Autobahnen dafür benutzt.
![grafik](https://github.com/devemux86/cruiser/assets/120722053/7f579feb-9ac4-48ec-b96f-6732c27049e2)
Umfahren von Oyonnax(2+3Kat).
Aufrufen kann man diese externen Profile so: Menue(3Striche links unten)/Routing/Optionen/Erweitert/BRouter Profil/Externes Profil öffnen-> externes Profil Laden.
(An den externen Profilen kann im Cruiser nichts mehr eingestellt werden.Allerdings könne sie mit einem Texteditor auf dem Smartphone verändert werden.Nach dem Ändern Cruiser ab und anschalten.)
Am besten im Menue 2 neue Routing Profile erstellen-

Diese BRouter-Profile lassen sich mit dem BRouter WebClient testen und verändern. Kleinere Werte bedeuten eine höhere Vermeidung
Damit lassen sich diese externen Profile leicht laden:
https://brouter.de/essbee/#map=7/48.9973/7.8259/OpenTopoMap
(Profiles/ein lokal gespeichertes Profil hochladen und aktivieren)

**Reply by WolfgangZ42** [reply]
Danke @Leeb64 !
Werde jetzt mal ein wenig damit "spielen" (und es mit Kurviger vergleichen).

**Comment by Leeb64** [comment]
Vorsicht!
Es gibt verschiedene Versionen des BRouter WebClient! Leider!Jeder benutzt da eine andere(ältere)BRouter Version.
Die Version die ziemlich das gleiche abliefert wie die im  Cruiser implementierte->:https://brouter.de/brouter-web/#map=6/48.8207/7.4377/OpenTopoMap
Diese Version ist nicht so komfortabel was das Einfügen der eigenen Profile angeht.
Man muss das Profil im WebClient löschen und das eigene einfügen.

**Reply by devemux86** [reply]
The different BRouter website versions are mentioned here:
- https://github.com/devemux86/cruiser/discussions/574#discussioncomment-9827856

**Reply by devemux86** [reply]
Unless playing with the new [environmental options](https://github.com/abrensch/brouter/blob/master/docs/developers/environmental_considerations_and_pseudo_tags.md), the routes should be similar.
